usecases/backup/coordinator.go

```diff
@@ -215,14 +215,12 @@ func (c *coordinator) canCommit(ctx context.Context, method Op) (map[string]stru
 			default:
 			}
 
-			// TODO: this is where nodeResolver can be used to find node hostname.
-			//       once found, it can be passed to pair below, rather than `node`
-			//host, found := c.nodeResolver.NodeHostname(node)
-			//if !found {
-			//	return fmt.Errorf("failed to find hostname for node %q", node)
-			//}
+			host, found := c.nodeResolver.NodeHostname(node)
+			if !found {
+				return fmt.Errorf("failed to find hostname for node %q", node)
+			}
 
-			reqChan <- pair{node, &Request{
+			reqChan <- pair{host, &Request{
```

Comment:
canCommit() is returning a set of host names as map[string]struct{} instead of node names.
QueryAll() relays on nodes to be a set of node names because it uses the node names to update the status for each node.

There is two possible solution for that:

1. Track the host name as well as node name and use former to make http calls and the other to maintain state.
2. Make the node resolver part of the client instead of the coordinator this way the coordinator doesn't need to care about resolving the node names.